exporters/otlp/connection.go

```diff
@@ -79,35 +140,136 @@ func (e *Exporter) indefiniteBackgroundConnection() {
 		// 2. Otherwise block until we are disconnected, and
 		//    then retry connecting
 		select {
-		case <-e.stopCh:
+		case <-oc.stopCh:
 			return
 
-		case <-e.disconnectedCh:
+		default:
+			// Not stopped right now, go wait for
+			// disconnect signal then.
+		}
+
+		select {
+		case <-oc.stopCh:
+			return
+
+		case <-oc.disconnectedCh:
 			// Normal scenario that we'll wait for
 		}
 
-		if err := e.connect(); err == nil {
-			e.setStateConnected()
+		if err := oc.connect(context.Background()); err == nil {
```

Comment:
Should this use the original context that the connection was started with instead?

Reply:
I don't think so. The original context might have some deadline set, which would be expired by the time the reconnection happens, which in turn would result in reconnection failure forever. The reconnection routine happens in background, so background context is appropriate here, IMO.